pkg/k8splan/watcher.go

```diff
@@ -182,17 +185,23 @@ func (w *watcher) start(ctx context.Context) {
 			if err != nil {
 				return secret, err
 			}
-			logrus.Debugf("[K8s] Calculated checksum to be %s", cp.Checksum)
+			logrus.Tracef("[K8s] Calculated checksum to be %s", cp.Checksum)
 
 			if secretChecksumData, ok := secret.Data[appliedChecksumKey]; ok {
 				secretChecksum := string(secretChecksumData)
-				logrus.Debugf("[K8s] Remote plan had an applied checksum value of %s", secretChecksum)
+				logrus.Tracef("[K8s] Remote plan had an applied checksum value of %s", secretChecksum)
 				if secretChecksum == cp.Checksum {
 					logrus.Debugf("[K8s] Applied checksum was the same as the plan from remote. Not applying.")
 					needsApplied = false
 				}
 			}
 
+			if !hasRunOnce {
```

Comment:
`hasRunOnce` is initialized before the `OnChange` handler is initialized/run.

This will only ever run once on start, as the variable is mutated within the block. The goal force a `needsApplied=true` at least once when a user `systemctl start/restart rancher-system-agent` for forced-reconciliation (or for events where a user reboots their node and loses temporary state).

FWIW, we only ever watch one specific secret (based on the configuration passed in) so I don't see a reason to build in multiplexing for "hasRunOnce" for multiple plans.